### PR TITLE
OAM-91: Added unit of orderable

### PR DESCRIPTION
**Pull request by tsznaj**

### CODE REVIEW

File: src/main/java/org/openlmis/referencedata/web/UnitOfOrderableController.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(UnitOfOrderableController.RESOURCE_PATH)
+public class UnitOfOrderableController extends BaseController {
```
[the remark this answers is not in the document]
Comment (tsznaj) [reply src/main/java/org/openlmis/referencedata/web/UnitOfOrderableController.java:48]: Good point, updated.

File: src/main/java/org/openlmis/referencedata/dto/UnitOfOrderableDto.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org.
+ */
```
[the remark this answers is not in the document]
Comment (tsznaj) [reply src/main/java/org/openlmis/referencedata/dto/UnitOfOrderableDto.java]: corrected

File: src/integration-test/java/org/openlmis/referencedata/web/UnitOfOrderableControllerIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+  private static final String DESCRIPTION = "description";
+  private static final int DISPLAY_ORDER = 1;
+  public static final String NEW_DESCRIPTION = "new description";
+
+  private UnitOfOrderable unitOfOrderable;
+  private UnitOfOrderableDto unitOfOrderableDto = new UnitOfOrderableDto();
+
+  private UUID unitOfOrderableId = UUID.randomUUID();
+
+
```
[the remark this answers is not in the document]
Comment (tsznaj) [reply src/integration-test/java/org/openlmis/referencedata/web/UnitOfOrderableControllerIntegrationTest.java]: corrected

File: src/main/java/org/openlmis/referencedata/domain/RightName.java [hunk 4]
```diff
@@ -25,6 +25,7 @@ public class RightName {
   public static final String SUPERVISORY_NODES_MANAGE = "SUPERVISORY_NODES_MANAGE";
   public static final String REQUISITION_GROUPS_MANAGE = "REQUISITION_GROUPS_MANAGE";
   public static final String ORDERABLES_MANAGE = "ORDERABLES_MANAGE";
+  public static final String UNIT_OF_ORDERABLES_MANAGE = "UNIT_OF_ORDERABLES_MANAGE";
```
[the remark this answers is not in the document]
Comment (tsznaj) [reply src/main/java/org/openlmis/referencedata/domain/RightName.java:28]: @sradziszewski I added this right - please check if it is correct.

File: src/main/java/org/openlmis/referencedata/repository/UnitOfOrderableRepository.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+package org.openlmis.referencedata.repository;
+
+import java.util.UUID;
+import org.openlmis.referencedata.domain.UnitOfOrderable;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UnitOfOrderableRepository extends JpaRepository<UnitOfOrderable, UUID> {}
```
[the remark this answers is not in the document]
Comment (tsznaj) [reply src/main/java/org/openlmis/referencedata/repository/UnitOfOrderableRepository.java]: Ok, added.

**Review by sradziszewski — APPROVED** [review]
all tests pass, apart from that the code looks ok

**Pull request merged**